simphony_mayavi/sources/slim_cuds_source.py

```diff
@@ -0,0 +1,543 @@
+        ----------
+        cuds : ABCParticles, ABCLattice, ABCMesh or H5Mesh
+            The CUDS dataset to be wrapped as VTK data source
+
+        point_scalars : str
+            CUBA name of the data to be selected as point scalars.
+            Default is the first available point scalars.
+
+        point_vectors : str
+            CUBA name of the data to be selected as point vectors.
+            Default is the first available point vectors.
+
+        cell_scalars : str
+            CUBA name of the data to be selected as cell scalars.
+            Default is the first available cell scalars.
+
+        cell_vectors : str
+            CUBA name of the data to be selected as cell vectors.
+            Default is the first available cell vectors.
+
+        Notes
+        -----
+        To turn off visualisation for a point/cell scalar/vector data,
+        assign the attribute to an empty string (i.e. point_scalars="")
+
+        Other optional keyword parameters are parsed to VTKDataSource.
+
+        Examples
+        --------
+        >>> cuds = Particles("test")
+
+        >>> # Say each particle has scalars "TEMPERATURE" and "MASS"
+        >>> # and vector data: "VELOCITY"
+        >>> cuds.add_particles([...])
+
+        >>> # Initialise the source and specify scalar data to visualise
+        >>> # but turn off the visualisation for point vectors
+        >>> source = SlimCUDSSource(
+                        cuds=cuds,
+                        point_scalars="MASS",
+                        point_vectors="")
+
+        >>> # Show it in Mayavi!
+        >>> from mayavi import mlab
+        >>> mlab.pipeline.glyph(source)
+        """
+        super(SlimCUDSSource, self).__init__(**traits)
+
+        if cuds:
+            # As a side effect, this populates all the
+            # scalar lists, the VTKCUDS, and the vtk_dataset
+            # (although by the nature of this class, it will be
+            # empty until data to show is chosen)
+            self.cuds = cuds
+
+        # Set the name to the user choice. If cuds is not defined,
+        # _point_scalars_list (etc.) is empty and nothing is selected.
+        if self._point_scalars_list and point_scalars is not None:
+            self.point_scalars_name = point_scalars
+
+        if self._point_vectors_list and point_vectors is not None:
+            self.point_vectors_name = point_vectors
+
+        if self._cell_scalars_list and cell_scalars is not None:
+            self.cell_scalars_name = cell_scalars
+
+        if self._cell_vectors_list and cell_vectors is not None:
+            self.cell_vectors_name = cell_vectors
+
+    # Public
+    # -------------------------------------------------------------------------
+
+    def start(self):
+        """This is invoked when this object is added to the mayavi
+        pipeline.
+        """
+        if self.running:
+            return
+
+        self._fill_datatype_enums()
+        self._update_vtk_cuds_from_cuds()
```

Comment:
What about making sure that `cuds` is set here?  Do we need to call `_fill_datatype_enums` and `_update_vtk_cuds_from_cuds` again given that they are called in `_set_cuds`?